src/cards/top-languages-card.js

```diff
@@ -200,6 +219,86 @@ const renderCompactLayout = (langs, width, totalLanguageSize) => {
   `;
 };
 
+/**
+ * @param {Lang[]} langs
+ * @param {number} width
+ * @param {number} height
+ * @param {number} totalLanguageSize
+ * @returns {string}
+ */
+const renderDonutLayout = (langs, width, height, totalLanguageSize) => {
+  const pie = d3.pie().value(function (d) {
+    return d.size;
+  });
+
+  const pieData = pie(langs);
+
+  const xPadding = 80;
+  const yPadding = 90;
+
+  const widthOffset = 50;
+  const heightOffset = 60;
+
+  const margin = 10;
+  const radius = (width - widthOffset - 2 * margin - yPadding) / 2;
+
+  const arc = d3
+    .arc()
+    .outerRadius(radius - 30)
+    .innerRadius(radius / 2);
+
+  const fakeDom = new JSDOM("<!DOCTYPE html><html><body></body></html>");
+  const body = d3.select(fakeDom.window.document).select("body");
```

Comment:
Not sure about this, Ideally I would like to keep the serverless functions light and dependency free. 

d3 and JSDOM both could tank the function's performance and lead to timeouts in vercel.

Reply:
In hindsight, it does make sense to have it all pure JS instead of using dependencies...did it the initial way because of the given example...I'll make the conversion

Reply:
@arndom Ah, great you are a hero! 💪🏻 Yea the example was just to show the feature. Sorry for the confusion. 😅

Reply:
No worries 👍